src/hooks/useAccounts.tsx

```diff
@@ -280,43 +146,18 @@ const useAccountsContext = () => {
     [getWalletFromSignature]
   );
 
-  useEffect(() => {
-    if (evmAddress) {
-      saveEvmDerivedAccount({ evmAddressInner: evmAddress, dydxAddress });
-    }
-  }, [evmAddress, dydxAddress]);
-
-  useEffect(() => {
-    if (solAddress) {
-      saveSolDerivedAccount({ solAddressInner: solAddress, dydxAddress });
-    }
-  }, [solAddress, dydxAddress]);
-
-  const signMessageAsync = useSignForWalletDerivation(connectedWallet);
+  const signMessageAsync = useSignForWalletDerivation(sourceAccount.walletInfo);
 
   useEffect(() => {
     (async () => {
-      if (connectedWallet?.connectorType === ConnectorType.Test) {
```

Comment:
I updated the address impersonation flow such that testFlag detection logic in `useWalletConnection` stores the WalletNetworkType.Test and the impersonated address together, so this useEffect here only needs to be responsible for deriving the local wallet